### PR TITLE
[v1.7.x] Message consumer Mock

### PR DESCRIPTION
**Pull request by gajosu**
Hey Mateus, I needed to test consumer messages, currently there are no mock methods for consumers, so I decided to implement a simple mock for consumer messages.

How is it used?
Using the `->addMockConsumerMessages()` method of the `Kafka` facade an array of Messages is configured, for example:
```php
Kafka::fake();
Kafka::addMockConsumerMessages(
    [
        new ConsumedMessage(
            topicName: 'test-topic',
            partition: 0,
            headers: [],
            body: ['test'],
            key: null,
            offset: 0,
            timestamp: 0
        ),
    ]
);
```

This will send the messages directly to the MessageHandler without connecting to the Kafka server.
If it is useful, I would appreciate ideas on how to improve it.

### CODE REVIEW

File: src/Support/Testing/Fakes/ConsumerFake.php [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+use Closure;
+use Junges\Kafka\Config\Config;
+use Junges\Kafka\Contracts\KafkaConsumerMessage;
+use Junges\Kafka\Contracts\MessageDeserializer;
+use Junges\Kafka\MessageCounter;
+use RdKafka\Conf;
+
+class ConsumerFake
+{
+
```
Comment (mateusjunges) [inline src/Support/Testing/Fakes/ConsumerFake.php:14]: This line should be removed

File: src/Support/Testing/Fakes/KafkaFake.php [hunk 2]
```diff
@@ -30,6 +34,52 @@ public function publishOn(string $topic, ?string $broker = null): ProducerBuilde
...
+        );
+    }
+
+    /**
+     * Add a mock messages to consume.
+     *
+     * @param \Junges\Kafka\Contracts\KafkaConsumerMessage|Junges\Kafka\Contracts\KafkaConsumerMessage[] $messages
+     * @return void
+     */
+    public function addMockConsumerMessages(KafkaConsumerMessage|array $messages): void
```
Comment (mateusjunges) [inline src/Support/Testing/Fakes/KafkaFake.php:62]: Maybe we can use `shouldReceive` or something similar, to keep consistency with laravel first party packages?

File: src/Support/Testing/Fakes/ConsumerBuilderFake.php [hunk 3]
```diff
@@ -0,0 +1,448 @@
...
+use Junges\Kafka\Config\BatchConfig;
+use Junges\Kafka\Config\NullBatchConfig;
+use Junges\Kafka\Consumers\CallableConsumer;
+use Junges\Kafka\Contracts\MessageDeserializer;
+use Junges\Kafka\Commit\Contracts\CommitterFactory;
+use Junges\Kafka\Exceptions\KafkaConsumerException;
+use Junges\Kafka\Support\Testing\Fakes\ConsumerFake;
+use Junges\Kafka\Contracts\HandlesBatchConfiguration;
+
+class ConsumerBuilderFake
```
Comment (mateusjunges) [inline src/Support/Testing/Fakes/ConsumerBuilderFake.php:19]: Can we extend this from `Junges\Kafka\Consumers\ConsumerBuilder` and change only the methods that needs a different behavior? I notice that this class is pretty much the same as the ConsumerBuilder.

**Review by mateusjunges — CHANGES_REQUESTED** [review]

File: src/Support/Testing/Fakes/KafkaFake.php (hunk 2 above)
Comment (gajosu) [reply src/Support/Testing/Fakes/KafkaFake.php:62]: Do you mean return an instance of Mockery?

Comment (mateusjunges) [reply src/Support/Testing/Fakes/KafkaFake.php:62]: No, I meant renaning the method to `shouldReceive`, or `shouldReceiveMessages`, something similar. Just a suggestion.

File: src/Support/Testing/Fakes/ConsumerFake.php [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+use Junges\Kafka\Contracts\KafkaConsumerMessage;
+use Junges\Kafka\Contracts\MessageDeserializer;
+use Junges\Kafka\MessageCounter;
+use RdKafka\Conf;
+
+class ConsumerFake
+{
+    private MessageCounter $messageCounter;
+    private bool $stopRequested = false;
+    private ?Closure $onStopConsume = null;
```
Comment (mateusjunges) [inline src/Support/Testing/Fakes/ConsumerFake.php:16]: I think we can also use PHP property promotion for this two variables. There is some other places where I need to fix it but no need to do it on this PR, just fix this one and it's fine.

**Review by mateusjunges — CHANGES_REQUESTED** [review]
Just a minor change and then I think we are good to go.

File: src/Support/Testing/Fakes/ConsumerFake.php (hunk 4 above)
Comment (gajosu) [reply src/Support/Testing/Fakes/ConsumerFake.php:16]: You mean this?

```php
public function __construct(
    private Config $config,
    private array $messages = [],
    private bool $stopRequested = false,
    private ?Closure $onStopConsume = null
) {
}
```

Comment (mateusjunges) [reply src/Support/Testing/Fakes/ConsumerFake.php:16]: yep

**Review by mateusjunges — APPROVED** [review]

**Comment by mateusjunges** [comment]
@gajosu I'll write the docs for this feature before releasing it.Thank you!

**Pull request merged**